src/display_row.jl

```diff
@@ -0,0 +1,75 @@
+export DisplayRow, render
+
+struct DisplayRow
+    size::Int
+    background::Char
+    context::Vector{Char}
+end
+
+function DisplayRow(len::Int; background=' ')
+    (textwidth(background) != 1) && (throw("Bad background char"))
+
+    return DisplayRow(len, background, fill(background, len))
+end
+
+Base.length(row::DisplayRow) = row.size
+
+Base.lastindex(row::DisplayRow) = row.size
+
+function get_element_index(row::DisplayRow, display_index::Int)
+    (textwidth(join(row.context)) < display_index) && (throw(BoundsError))
+
+    accumulate_width = index = pre = post = 0
+    for c in row.context
+        width = textwidth(c)
+        accumulate_width += width
+        index += 1
+        if accumulate_width >= display_index
+            pre = display_index - (accumulate_width-width) - 1
+            post = accumulate_width-display_index
+            break
+        end
+    end
+
+    return (i=index, pre=pre, post=post)
+end
+
+function Base.deleteat!(row::DisplayRow, display_index::Int)
+    index = get_element_index(row, display_index)
+    deleteat!(row.context, index.i)
+    for i=1:(index.pre+index.post+1)
+        insert!(row.context, index.i, row.background)
+    end
```

Comment:
minor advice: this loop can be remove by array operations.
for example
```jl
len = index.pre + index.post + 1
old_len = length(row.context)
new_len = old_len + len
resize!(row.context, new_len)
copyto!(row.context, index.i + len, row.context, index.i, old_len - index.i + 1)
fill!(@view(row.context[index.i:index.i+len]), row.background)
```
(correctness need to be checked)

Reply:
But it seems more complicated than original one.

Reply:
the original one is much slow since it you need to realloc on each `insert!`

Reply:
If it is the way to go, just extract out a function to operate on array.